src/Sqids.php

```diff
@@ -772,40 +769,36 @@ public function decode(string $id): array
 
     protected function shuffle(string $alphabet): string
     {
-        $chars = str_split($alphabet);
-
-        for ($i = 0, $j = count($chars) - 1; $j > 0; $i++, $j--) {
-            $r = ($i * $j + ord($chars[$i]) + ord($chars[$j])) % count($chars);
-            [$chars[$i], $chars[$r]] = [$chars[$r], $chars[$i]];
+        for ($i = 0, $j = strlen($alphabet) - 1; $j > 0; $i++, $j--) {
+            $r = ($i * $j + ord($alphabet[$i]) + ord($alphabet[$j])) % strlen($alphabet);
+            [$alphabet[$i], $alphabet[$r]] = [$alphabet[$r], $alphabet[$i]];
         }
 
-        return implode('', $chars);
+        return $alphabet;
     }
 
     protected function toId(int $num, string $alphabet): string
     {
-        $id = [];
-        $chars = str_split($alphabet);
-
-        $result = $num;
-
+        $id = '';
         do {
-            array_unshift($id, $chars[$this->math->intval($this->math->mod($result, count($chars)))]);
-            $result = $this->math->divide($result, count($chars));
-        } while ($this->math->greaterThan($result, 0));
+            $id = $alphabet[$this->math->intval($this->math->mod($num, strlen($alphabet)))] . $id;
```

Comment:
Appending to the end of the string is the same as using `array_unshift` on an array.